### DISCUSSION TITLE
How to use Lambdad to save hovernet instance to csv along with file name

### DISCUSSION THREAD

**Discussion by OmarAshkar** [Project-MONAI/MONAI#6322] [Q&A] [answered]
Hi,

I am trying to save hovernet instance_info to csv file. Currently I have the following simple function.
```
    def inst_info_to_df(instance_info):
        df = pd.DataFrame.from_dict(instance_info,orient='index')
        df.to_csv("slide_name.csv")
        print("done export")
```
I just used `Lambdad(keys=["instance_info"], func=inst_info_to_df)`. 

This works, but I need to also pass file names from `image_meta_dict`. I cannot find a way to pass both instance_info and image_meta_dict.

Thanks in advance !

**Comment by KumoLiu** [comment, marked as answer]
Hi @OmarAshkar, you could use `Lambda`
```
    def inst_info_to_df(data):
        df = pd.DataFrame.from_dict(data["instance_info"],orient='index')
        df.to_csv(f"slide_name{data['image_meta_dict']['filename_or_obj'].split('/')[-1]}.csv")
        return data

Lambda(func=lambda x: inst_info_to_df(x))
```

Hope it can help you, thanks!
